### DISCUSSION TITLE
Synpress - Cypress and Playwright with metamask support for e2e testing of dapps

### DISCUSSION THREAD

**Discussion by drptbl** [cypress-io/cypress#14164] [General]
Hey :wave:,

[Synpress](https://github.com/Synthetixio/synpress) is e2e testing framework based on [Cypress](https://github.com/cypress-io/cypress) and [Playwright](https://playwright.dev/) with support for [MetaMask](https://metamask.io/).

Synpress makes sure to always use latest version of metamask and puts a lot of effort to make sure that dapp tests are stable and trustful.

It also provides an easy way to use and access metamask straight from your e2e tests with all features of cypress and playwright.

🔥 Synpress works out-of-the-box with other frameworks! There is no need to use it directly. Check [usage examples](https://github.com/Synthetixio/synpress/#usage-examples) for more details.

## Supported frameworks

- [synpress](https://github.com/Synthetixio/synpress) (standalone, uses cypress under-the-hood)
- [Cypress](https://github.com/cypress-io/cypress) (as a plugin)
- [Playwright](https://playwright.dev/) (as a plugin)

## Supported wallets

- [MetaMask](https://metamask.io/)

## Usage examples

- ⭐ [synpress-examples](https://github.com/drptbl/synpress-examples/)
- [synpress](https://github.com/Synthetixio/synpress/tree/dev/tests/e2e)

## Links
- GitHub:
https://github.com/Synthetixio/synpress
- npm:
https://www.npmjs.com/package/@synthetixio/synpress

I hope you will find it useful.

Cheers,
Jakub.

**Comment by f1lander** [comment]
@drptbl Hi! I'm responsible to build this, https://goerli.launchpad.obol.tech/ is dapp for distributed validators, I want to integrate synpress with this but I'm not getting success, I found the documentation pretty hard and have no example to integrate it into a react application, could you help me?

**Reply by drptbl** [reply]
Hey @f1lander,

This could be helpful:
https://github.com/drptbl/synpress-examples

Thanks,
Jakub.
